lib/parsers/jpeg_parser.rb

```diff
@@ -66,16 +67,24 @@ def scan
 
     Measurometer.add_distribution_value('format_parser.JPEGParser.bytes_read_until_capture', @buf.pos)
 
+    # A single file might contain multiple EXIF data frames. In a JPEG this would
+    # manifest as multiple APP1 markers. The way different programs handle these
+    # differs, for us it makes the most sense to simply "flatten" them top-down.
+    # So we start with the first EXIF frame, and we then allow the APP1 markers
+    # that come later in the file to override the properties they _do_ specify.
+    flat_exif = FormatParser::EXIFParser::EXIFStack.new(@exif_data_frames)
+
     # Return at the earliest possible opportunity
     if @width && @height
+      dw, dh = flat_exif.rotated? ? [@height, @width] : [@width, @height]
```

Comment:
What does `flat_exif.rotated?` return if the images is rotated 180 degrees?

Reply:
`false`. We could name it something like `orientation_changed?` instead, but I didn't bother because the method is not exposed to the outside from the format parser `Result` object.